kms-message/src/kms_kmip_response_parser_private.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef KMS_KMIP_RESPONSE_PARSER_PRIVATE_H
+#define KMS_KMIP_RESPONSE_PARSER_PRIVATE_H
+
+#include "kms_message/kms_response.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+
+/* kms_kmip_response_parser_t is a private type used for parsing a KMIP
+ * response. */
+typedef struct _kms_kmip_response_parser_t kms_kmip_response_parser_t;
+
+/* KMS_KMIP_RESPONSE_PARSER_FIRST_LENGTH is the number of bytes needed to
+ * determine the length of the remaining message. */
+#define KMS_KMIP_RESPONSE_PARSER_FIRST_LENGTH 8
+
+int32_t
+kms_kmip_response_parser_wants_bytes (const kms_kmip_response_parser_t *parser,
+                                      int32_t max);
+
+bool
+kms_kmip_response_parser_feed (kms_kmip_response_parser_t *parser,
+                               const uint8_t *buf,
+                               uint32_t len);
+
+/* kms_kmip_response_parser_get_response returns a kms_response_t and resets the
+ * parser. */
```

Comment:
Would a user ever want to request a (copy of the) response _without_ resetting the parser state?

Reply:
Then I am fine with leaving as-is. Separate functions could easily be added on-demand.